core/src/test/scala/org/apache/spark/ui/UISuite.scala

```diff
@@ -101,6 +125,25 @@ class UISuite extends FunSuite {
     }
   }
 
+  test("jetty with https binds to port 0 correctly") {
+    val sparkConf = new SparkConf()
+      .set("spark.ui.https.enabled", "false")
+      .set("spark.ui.https.keyStore", "./src/test/resources/spark.keystore")
+      .set("spark.ui.https.keyStorePassword", "123456")
+      .set("spark.ui.https.keyPassword", "123456")
+    val serverInfo = JettyUtils.startJettyServer(
+      "0.0.0.0", 0, Seq[ServletContextHandler](), sparkConf)
+    val server = serverInfo.server
+    val boundPort = serverInfo.boundPort
+    assert(server.getState === "STARTED")
+    assert(boundPort != 0)
+    Try { new ServerSocket(boundPort) } match {
```

Comment:
Using `intercept` here might be cleaner. And in case of failure, you need to close the server socket you're creating.